packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/Form/reducer.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable consistent-return */
+/*
+ *
+ * ApplicationInfosPage Form reducer
+ *
+ */
+
+import produce from 'immer';
+
+const initialState = {
+  menuLogo: null,
+};
+
+const reducer = (state = initialState, action) =>
+  produce(state, draftState => {
+    switch (action.type) {
+      case 'SET_CUSTOM_MENU_LOGO': {
+        draftState.menuLogo = action.value;
+        break;
+      }
+      default: {
+        return draftState;
+      }
+    }
+  });
+
+export default reducer;
+export { initialState };
```

Comment:
Same question here, do tou really need all these things to handle that state?

Reply:
I mean, do you really need:

- Use useReducer hook to handle one value `menuLogo`
- Import and use immer
- Initial state file
- reducer test file

This seems to add a lot of noise in the codebase for not much 😅  what do you think?

Reply:
@HichamELBSI I understand your point of view, it was my first thought too 😅
For the Form component I actually need a state init since the state will be based on `projectSettingsStored` data coming from our back-end. It would be doable with only `useState` indeed but @soupette recommended me to go with `useReducer`. Plus, at least for the Form component, it should soon handle more than menuLogo.
For LogoInput though, maybe a `useState` would be enough 🤷🏻‍♀️
WDYT?

Reply:
Make sense :)